drivers/SmartThings/matter-evse/src/init.lua

```diff
@@ -189,105 +259,107 @@ local function create_poll_report_schedule(device)
+      local comp_id = "importedEnergy"
+      if #evse_eps > 0 then
+        comp_id = "main"
+      end
+      report_energy_to_app(device, comp_id, cumulative_energy_imported, startTime, endTime)
+    end
+
+    -- If energy exported is set, it must be for Solar Power / Battery Storage Device.
+    if cumulative_energy_exported ~= nil then
+      report_energy_to_app(device, "exportedEnergy", cumulative_energy_exported, startTime, endTime)
+    end
   end, "polling_report_schedule_timer")
 
   device:set_field(RECURRING_REPORT_POLL_TIMER, timer)
 end
 
-local function create_poll_schedules_for_cumulative_energy_imported(device)
+local function create_poll_schedules_for_cumulative_energy_reports(device)
   if not device:supports_capability(capabilities.powerConsumptionReport) then
     return
   end
   create_poll_schedule(device)
   create_poll_report_schedule(device)
 end
 
+local function delete_reporting_timer(device)
+  local reporting_poll_timer = device:get_field(RECURRING_REPORT_POLL_TIMER)
+  if reporting_poll_timer ~= nil then
+    device.thread:cancel_timer(reporting_poll_timer)
+    device:set_field(RECURRING_REPORT_POLL_TIMER, nil)
+  end
+end
+
 local function delete_poll_schedules(device)
   local poll_timer = device:get_field(RECURRING_POLL_TIMER)
-  local reporting_poll_timer = device:get_field(RECURRING_REPORT_POLL_TIMER)
   if poll_timer ~= nil then
     device.thread:cancel_timer(poll_timer)
     device:set_field(RECURRING_POLL_TIMER, nil)
   end
-  if reporting_poll_timer ~= nil then
-    device.thread:cancel_timer(reporting_poll_timer)
-    device:set_field(RECURRING_REPORT_POLL_TIMER, nil)
-  end
+  delete_reporting_timer(device)
 end
 
 -- Lifecycle Handlers --
 local function device_init(driver, device)
   device:subscribe()
   device:set_endpoint_to_component_fn(endpoint_to_component)
   device:set_component_to_endpoint_fn(component_to_endpoint)
-  create_poll_schedules_for_cumulative_energy_imported(device)
+  create_poll_schedules_for_cumulative_energy_reports(device)
   local current_time = os.time()
   local current_time_iso8601 = epoch_to_iso8601(current_time)
   -- emit current time by default
   device:emit_event(capabilities.evseChargingSession.targetEndTime(current_time_iso8601))
 end
 
 local function device_added(driver, device)
-  local electrical_sensor_eps = get_endpoints_for_dt(device, ELECTRICAL_SENSOR_DEVICE_ID) or {}
-  local device_energy_mgmt_eps = get_endpoints_for_dt(device, DEVICE_ENERGY_MANAGEMENT_DEVICE_ID) or {}
-  local component_to_endpoint_map = {
-    ["main"] = device.MATTER_DEFAULT_ENDPOINT,
-    ["electricalSensor"] = electrical_sensor_eps[1],
-    ["deviceEnergyManagement"] = device_energy_mgmt_eps[1]
-  }
-  log.debug("component_to_endpoint_map " .. utils.stringify_table(component_to_endpoint_map))
-  device:set_field(COMPONENT_TO_ENDPOINT_MAP, component_to_endpoint_map, { persist = true })
+  local evse_eps = get_endpoints_for_dt(device, EVSE_DEVICE_TYPE_ID)
```

Comment:
```suggestion
  local evse_eps = get_endpoints_for_dt(device, EVSE_DEVICE_TYPE_ID) or {}
```
To prevent an error on the next line in the case that `get_endpoints_for_dt` returns nil